pkg/ovs/ovsctl/ofctl_test.go

```diff
@@ -0,0 +1,237 @@
+	assert.Equal(t, expectedFlows, out)
+}
+
+func TestDumpGroups(t *testing.T) {
+	client := &ovsCtlClient{
+		bridge: "br-int",
+	}
+	runCommand = func(cmd string) ([]byte, error) {
+		var x = []byte{}
+		for i := 0; i < len(testDumpGroups); i++ {
+			b := []byte(testDumpGroups[i])
+			for j := 0; j < len(b); j++ {
+				x = append(x, b[j])
+			}
+			x = append(x, "\n"...)
+		}
+		return x, nil
+	}
+	defer func() {
+		runCommand = runCmd
+	}()
+	out, err := client.DumpGroups()
+	assert.NoError(t, err)
+	expectedGroups := []string{"group_id=1,type=all,bucket=mod_dl_src=00:00:00:11:11:11,mod_dl_dst=00:00:00:22:22:22,output:2,bucket=mod_dl_src=00:00:00:11:11:11,mod_dl_dst=00:00:00:22:22:22,output:3", "group_id=2,type=indirect,bucket=mod_dl_src=00:00:00:99:11:11,mod_dl_dst=00:00:00:99:22:22,output:2", "group_id=3,type=select,bucket=bucket=output:1,output:2,bucket=output:3,bucket=output:4", "group_id=4,type=ff,bucket=watch_port:2,watch_group:2,output:3,watch_port:3,watch_group:3,output:4"}
+	assert.Equal(t, expectedGroups, out)
+}
+
+func TestDumpGroup(t *testing.T) {
+	client := &ovsCtlClient{
+		bridge: "br-int",
+	}
+	groupID := 3
+	runCommand = func(cmd string) ([]byte, error) {
+		var x = []byte{}
+		id := cmd[strings.Index(cmd, "br-int")+7 : strings.Index(cmd, " -O")]
+		for i := 0; i < len(testDumpGroups); i++ {
+			if !strings.Contains(testDumpGroups[i], id) {
+				continue
+			}
+			b := []byte(testDumpGroups[i])
+
+			for j := 0; j < len(b); j++ {
+				x = append(x, b[j])
+			}
+			x = append(x, "\n"...)
+
+		}
+		return x, nil
+	}
+	defer func() {
+		runCommand = runCmd
+	}()
+	out, err := client.DumpGroup(uint32(groupID))
+	assert.NoError(t, err)
+	expectedGroup := "group_id=3,type=select,bucket=bucket=output:1,output:2,bucket=output:3,bucket=output:4"
+	assert.Equal(t, expectedGroup, out)
+}
+
+func TestDumpPortsDesc(t *testing.T) {
+	client := &ovsCtlClient{
+		bridge: "br-int",
+	}
+	runCommand = func(cmd string) ([]byte, error) {
+		var x = []byte{}
+		for i := 0; i < len(testDumpPortsDesc); i++ {
+			b := []byte(testDumpPortsDesc[i])
+			for j := 0; j < len(b); j++ {
+				x = append(x, b[j])
+			}
+			x = append(x, "\n"...)
+		}
+		return x, nil
+	}
+	defer func() {
+		runCommand = runCmd
+	}()
+	out, err := client.DumpPortsDesc()
+	assert.NoError(t, err)
+	expectedDesc := [][]string([][]string{{"1(p2p1): addr:b8:59:9f:d2:1c:ba", "     config:     0", "     state:      0", "    current:    AUTO_NEG", "     advertised: 1GB-FD 10GB-FD AUTO_NEG", "    supported:  1GB-FD 10GB-FD AUTO_NEG AUTO_PAUSE", "     speed: 0 Mbps now, 10000 Mbps max"}, {"2(vnet0): addr:fe:54:00:11:8f:ea", "     config:     0", "     state:      0", "     current:    10MB-FD COPPER", "    speed: 10 Mbps now, 0 Mbps max"}, {"LOCAL(ovs_pvp_br0): addr:b8:59:9f:d2:1c:ba", "     config:     PORT_DOWN", "     state:      LINK_DOWN", "     speed: 0 Mbps now, 0 Mbps max"}})
+	assert.Equal(t, expectedDesc, out)
```

Comment:
when using asset multiple times use assert := assert.New(t) in beginning to avoid using t at multiple places